File: apps/websocket-skyblock/src/mongodb/collections/AuctionItemsAll.cs
using System.Runtime.Serialization;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

public class AuctionItemsAll
{
    [BsonId]
    [DataMember]
    public ObjectId _id { get; set; }
    /// <summary>
    /// The Hypixel ID of the item.
    /// </summary>
    public string ID;
    /// <summary>
    /// The name of the item.
    /// </summary>
    public string Name;
    /// <summary>
    /// List of "Name" of the extra attributes that this item has been seen with.
    /// </summary>
    public List<string> AuctionTags = new List<string>();

    public AuctionItemsAll(string _ID, string _Name, List<string>? _AuctionTags = null)
    {
        ID = _ID;
        Name = _Name;

        if (_AuctionTags != null)
            AuctionTags = _AuctionTags;
    }
}

File: apps/discord-bot/app/src/main/java/discordbot/mongodb/collections/AuctionItemsAll.java
package discordbot.mongodb.collections;

import java.util.ArrayList;

import org.bson.codecs.pojo.annotations.BsonId;
import org.bson.types.ObjectId;

public class AuctionItemsAll {
    @BsonId
    public ObjectId _id;
    /**
     * The Hypixel ID of the item.
     */
    public String ID;
    /**
     * The name of the item.
     */
    public String Name;
    /**
     * List of "Name" of the extra attributes that this item has been seen with.
     */
    public ArrayList<String> AuctionTags = new ArrayList<String>();

    public AuctionItemsAll(String _ID, String _Name) {
        this(_ID, _Name, new ArrayList<String>());
    }

    public AuctionItemsAll(String _ID, String _Name, ArrayList<String> _AuctionTags) {
        ID = _ID;
        Name = _Name;
        AuctionTags = _AuctionTags;
    }
}
